app/models/planting.rb

```diff
@@ -64,17 +64,17 @@ def finished_must_be_after_planted
   end
 
   def planting_slug
-    "#{owner.login_name}-#{garden}-#{crop}".downcase.gsub(' ', '-')
+    "#{owner.login_name}-#{garden.name}-#{crop.name}".downcase.gsub(' ', '-')
```

Comment:
OK, I see what's going on - the test is now breaking because we're looking up the `name` field on a nonexistent crop. Previously, we'd create a slug `pozorvlak-allotment-nil`, which was fine, and then we'd throw it away because the planting was invalid. So I guess we need to move the test to before the slug is generated, or make the slug-generation more robust.